homeassistant/components/launch_library/sensor.py

```diff
@@ -26,59 +32,55 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Create the launch sensor."""
-
     name = config[CONF_NAME]
-
     session = async_get_clientsession(hass)
-    launches = Launches(hass.loop, session)
-    sensor = [LaunchLibrarySensor(launches, name)]
-    async_add_entities(sensor, True)
+    launches = PyLaunches(session)
+
+    async_add_entities([LaunchLibrarySensor(launches, name)], True)
 
 
 class LaunchLibrarySensor(Entity):
     """Representation of a launch_library Sensor."""
 
-    def __init__(self, launches, name):
+    def __init__(self, launches: PyLaunches, name: str) -> None:
         """Initialize the sensor."""
         self.launches = launches
-        self._attributes = {}
+        self.next_launch = None
         self._name = name
-        self._state = None
 
-    async def async_update(self):
+    async def async_update(self) -> None:
         """Get the latest data."""
-        await self.launches.get_launches()
-        if self.launches.launches is None:
-            _LOGGER.error("No data received")
-            return
         try:
-            data = self.launches.launches[0]
-            self._state = data["name"]
-            self._attributes["launch_time"] = data["start"]
-            self._attributes["agency"] = data["agency"]
-            agency_country_code = data["agency_country_code"]
-            self._attributes["agency_country_code"] = agency_country_code
-            self._attributes["stream"] = data["stream"]
-            self._attributes[ATTR_ATTRIBUTION] = ATTRIBUTION
-        except (KeyError, IndexError) as error:
-            _LOGGER.debug("Error getting data, %s", error)
+            launches = await self.launches.upcoming_launches()
+            if launches:
+                self.next_launch = launches[0]
+        except PyLaunchesException as exception:
+            _LOGGER.error("Error getting data, %s", exception)
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the sensor."""
         return self._name
 
     @property
-    def state(self):
+    def state(self) -> Optional[str]:
         """Return the state of the sensor."""
-        return self._state
+        if self.next_launch:
+            return self.next_launch.name
 
```

Comment:
Please return explicitly in all cases if we return a value anywhere. `return None` in the other case.

Reply:
Default is `return None`?

Reply:
That's implicit. We should return a value explicitly at all exit points of the function whenever we return a value anywhere in the function.